### ISSUE TITLE
bug: CapacitorHttp.request never returns for unreachable servers (iOS)

### ISSUE THREAD

**Issue by zefhemel** [ionic-team/capacitor#6236]
### Capacitor Version
```
💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 4.6.2
  @capacitor/core: 4.6.2
  @capacitor/android: 4.6.2
  @capacitor/ios: 4.6.2

Installed Dependencies:

  @capacitor/cli: 4.6.1
  @capacitor/android: 4.6.1
  @capacitor/core: 4.6.1
  @capacitor/ios: 4.6.1

[success] iOS looking great! 👌
```

### Platform(s)
iOS (but not tested on Android yet)

### Current Behavior
```
try {
    console.log("Sending request");
    const resp = await CapacitorHttp.request({
      url: "http://localhost:3002",
      readTimeout: 1000,
      connectTimeout: 1000,
    });
    console.log("Response", resp);
  } catch (e: any) {
    console.error("ERROR", e);
  }
```

Assuming that `localhost:3002` is not actually reachable (can be any non-reachable server) what I see in the iOS simulator logs now is:

```
⚡️  [log] - Sending request
⚡️  To Native ->  CapacitorHttp request 43313132
2023-01-20 15:07:36.143108+0100 SilverBullet[38280:9724338] [connection] nw_socket_handle_socket_event [C1.1.1:2] Socket SO_ERROR [61: Connection refused]
2023-01-20 15:07:36.144018+0100 SilverBullet[38280:9724338] [connection] nw_socket_handle_socket_event [C1.1.2:2] Socket SO_ERROR [61: Connection refused]
2023-01-20 15:07:36.144812+0100 SilverBullet[38280:9724338] Connection 1: received failure notification
2023-01-20 15:07:36.145062+0100 SilverBullet[38280:9724338] Connection 1: failed to connect 1:61, reason -1
2023-01-20 15:07:36.145113+0100 SilverBullet[38280:9724338] Connection 1: encountered error(1:61)
2023-01-20 15:07:36.145774+0100 SilverBullet[38280:9724338] Task <2D5EEC72-607D-4827-A435-1CCCE5755803>.<1> HTTP load failed, 0/0 bytes (error code: -1004 [1:61])
2023-01-20 15:07:36.148997+0100 SilverBullet[38280:9724342] Task <2D5EEC72-607D-4827-A435-1CCCE5755803>.<1> finished with error [-1004] Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={_kCFStreamErrorCodeKey=61, NSUnderlyingError=0x600003e09f80 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_NSURLErrorNWPathKey=satisfied (Path is satisfied), interface: en0[802.11], _kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <2D5EEC72-607D-4827-A435-1CCCE5755803>.<1>, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <2D5EEC72-607D-4827-A435-1CCCE5755803>.<1>"
), NSLocalizedDescription=Could not connect to the server., NSErrorFailingURLStringKey=http://localhost:3002/, NSErrorFailingURLKey=http://localhost:3002/, _kCFStreamErrorDomainKey=1}
2023-01-20 15:07:36.149091+0100 SilverBullet[38280:9724342] [NSURLSession sharedSession] may not be invalidated
```

So clearly Capacitor itself is detecting that it cannot reach this URL, however it never makes it back into "JS land". Neither the catch, nor the console.log("Response", ...) line ever gets called.

### Expected Behavior
Either throw an exception or return a response with some response error code, either would work (exception would make more sense, probably).

### Code Reproduction
See the code listed above

### Other Technical Details

`npm --version` output:
8.12.1

`node --version` output:
v18.4.0

`pod --version` output (iOS issues only):
1.11.3

**Comment by jcesarmobile** [comment]
this was fixed in 4.6.2

**Comment by zefhemel** [comment]
And here I was so sure I had upgraded to that, but indeed not properly. Confirmed, this is fixed. Sorry for this!
